azure/cluster.tf

```diff
@@ -8,8 +8,11 @@ resource "azurerm_linux_virtual_machine" "redpanda" {
   count                        = var.vm_instances
   resource_group_name          = azurerm_resource_group.redpanda.name
   location                     = azurerm_resource_group.redpanda.location
-  availability_set_id          = azurerm_availability_set.redpanda.id
+  availability_set_id          = var.ha ? null : azurerm_availability_set.redpanda.0.id
```

Comment:
So if `ha` is `true`, then we don't create an availability set? That seems opposite of what we want right?

Reply:
So the _old_ behaviour for this provider is that we would automatically deploy machines in an availability set, presumably for resilience. However availability sets don't give the ability to introspect which machines are placed into which fault domain, so there is limited use in terms of rack awareness in Redpanda. We _could_ remove availability sets altogether, other than it would be a change in behaviour from what was there before.

When `ha=true` we now deploy into a flexible scale set (with three fault domains) which then gives us information about which fault domain each VM is, that we can use for rack awareness. 

I figured there was no harm in leaving the existing behaviour in there, but happy to take advice.

Reply:
Thanks, this helps explain the use of availability sets in this project. Scale set (rather than availability set) is the vehicle for enabling rack awareness, and I was mistakenly trying to understand how the availability set could provide that especially given changes in this PR.